code/game/objects/items/devices/radio/radio.dm

```diff
@@ -186,12 +186,31 @@
 				. = TRUE
 
 /obj/item/radio/talk_into(atom/movable/M, message, channel, list/spans, datum/language/language, list/message_mods)
-	if(!spans)
-		spans = list(M.speech_span)
-	if(!language)
-		language = M.get_selected_language()
-	INVOKE_ASYNC(src, .proc/talk_into_impl, M, message, channel, spans.Copy(), language, message_mods)
-	return ITALICS | REDUCE_RANGE
+	if(HAS_TRAIT(M, TRAIT_SIGN_LANG)) //Forces Sign Language users to wear the translation gloves to speak over radios
+		var/mob/living/carbon/mute = M
+		if(istype(mute))
+			var/empty_indexes = mute.get_empty_held_indexes() //How many hands the player has empty
+			var/obj/item/clothing/gloves/radio/G = mute.get_item_by_slot(ITEM_SLOT_GLOVES)
+			if(!istype(G))
+				return FALSE
+			else
+				if(length(empty_indexes) == 1)
+					message = stars(message)
+				if(length(empty_indexes) == 0) //Due to the requirement of gloves, the arm check for normal speech would be redundant here.
+					return FALSE
+				if(!spans)
+					spans = list(M.speech_span)
+				if(!language)
+					language = M.get_selected_language()
+				INVOKE_ASYNC(src, .proc/talk_into_impl, M, message, channel, spans.Copy(), language, message_mods)
+				return ITALICS | REDUCE_RANGE
+	else
```

Comment:
cant this just be removed and let it do the other part of the proc after finishing the if

Reply:
True, I will fix this asap